Nodejs/Product/Nodejs/NpmUI/NpmPackageInstallViewModel.cs

```diff
@@ -357,17 +367,64 @@ public PackageCatalogEntryViewModel SelectedPackage
             }
         }
 
+        public bool IsExecutingCommand
+        {
+            get
+            {
+                return this.isExecutingCommand;
+            }
+            set
+            {
+                this.isExecutingCommand = value;
+
+                OnPropertyChanged();
+                OnPropertyChanged("ExecutionProgressVisibility");
+            }
+        }
+
+        public Visibility ExecutionProgressVisibility
+        {
+            get { return IsExecutingCommand ? Visibility.Visible : Visibility.Collapsed; }
+        }
+
         #endregion
 
         public event PropertyChangedEventHandler PropertyChanged;
 
         protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
         {
-            var handler = this.PropertyChanged;
-            if (handler != null)
+            this.PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposed)
             {
-                handler(this, new PropertyChangedEventArgs(propertyName));
+                if (disposing)
+                {
+                    this.filterTimer?.Dispose();
+                }
+
+                var controller = this.npmController;
+                if (controller != null)
+                {
+                    this.npmController.FinishedRefresh -= this.NpmController_FinishedRefresh;
```

Comment:
>this.npmController [](start = 20, length = 18)

`controller`.  Was it extracted for fear of a race?

Reply:
no the compiler complained

Reply:
mostly worried about the VS shutdown/solution close scenario